python/ucxx/ucxx/_lib/libucxx.pyx

```diff
@@ -95,9 +95,20 @@ def _get_host_buffer(uintptr_t recv_buffer_ptr):
     return np.asarray(HostBufferAdapter._from_host_buffer(host_buffer))
 
 
-cdef shared_ptr[Buffer] _rmm_am_allocator(size_t length):
-    cdef shared_ptr[RMMBuffer] rmm_buffer = make_shared[RMMBuffer](length)
-    return dynamic_pointer_cast[Buffer, RMMBuffer](rmm_buffer)
+# A pure C++ is used here for the allocator, a Cython function always raises
+# exceptions back to Python, but the allocator here is executed as a callback
+# directly in C++, therefore exceptions have to be raised to the C++ caller
+# directly.
+cdef extern from *:
+    """
+    #include <memory>
+    #include <ucxx/buffer.h>
+
+    std::shared_ptr<ucxx::Buffer> _rmm_am_allocator(size_t length) {
+        return std::make_shared<ucxx::RMMBuffer>(length);
+    }
+    """
+    void _rmm_am_allocator(int) except+*
```

Comment:
```suggestion
    void _rmm_am_allocator(size_t) except+*
```
?

Reply:
Good catch, not sure why I wrote `int` here. Anyways, this was reverted and fixed in https://github.com/rapidsai/ucxx/pull/315/commits/bf54be9367a7d91ffe15a14bd0059954f223f6fa as well.